### PR TITLE
Add BZIP2 compression algorithm

### PR DESCRIPTION
**Pull request by Luke-Shepp**
Creates a new compression class to support BZIP2 compression and decompression, with accompanied unit test.

```
➜  utopia-php-storage git:(feat-bzip2-compression) ✗ ./vendor/bin/phpunit --filter BZIP2Test
PHPUnit 9.5.25 by Sebastian Bergmann and contributors.

....                                                                4 / 4 (100%)

Time: 00:01.870, Memory: 17.55 MB

OK (4 tests, 12 assertions)
```

Part 1 of https://github.com/appwrite/appwrite/issues/3996

### CODE REVIEW

File: tests/Storage/Compression/Algorithms/BZIP2Test.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Utopia\Tests;
```
Comment (stnguyen90) [inline tests/Storage/Compression/Algorithms/BZIP2Test.php:3]: Would you please use:

```suggestion
namespace Tests\Storage\Compression\Algorithms;
```

to match psr-4 standards?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
@Luke-Shepp Great PR! 🤯 We left some comments during the review, please check them out.

**Comment by stnguyen90** [comment]
Also, it looks like the tests are failing:

image

**Comment by christyjacob4** [comment]
Please sync your branch with `main` . It has fixes for the failing CI tests

**Comment by Meldiron** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Pull request closed without merging**